.github/workflows/pull_request.yml

```diff
@@ -41,11 +41,14 @@ jobs:
       - name: build-all target
         run: make build-all
 
-      - name: Archive binary
-        uses: actions/upload-artifact@v2
-        with:
-          name: receptor
-          path: receptor
+      - name: Upload binary
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          pip install boto3
+          ansible -i localhost, -c local all -m aws_s3 \
+            -a "bucket=receptor-nightlies object=PRs/${{github.event.number}}/receptor src=./receptor mode=put"
```

Comment:
Doesn't this also allow any github user to upload arbitrary source code to be build by actions and uploaded into the bucket?